src/org/etools/j1939_84/bus/j1939/packets/Slot.java

```diff
@@ -19,197 +18,318 @@
+        Map<Integer, Slot> slots = new HashMap<>();
+        String[] values;
+
+        final InputStream is = Resources.class.getResourceAsStream("slots.csv");
+        final InputStreamReader isReader = new InputStreamReader(is, StandardCharsets.ISO_8859_1);
+        try (CSVReader reader = new CSVReader(isReader)) {
+            while ((values = reader.readNext()) != null) {
+                final int id = Integer.parseInt(values[0]);
+                final String name = values[1];
+                final String type = values[2];
+                final Double scaling = doubleValue(values[3]);
+                final String unit = values[4];
+                final Double offset = doubleValue(values[5]);
+                final int length = intValue(values[6], -1);
+                Slot slot = new Slot(id, name, type, scaling, offset, unit, length);
+                slots.put(id, slot);
+            }
+        } catch (Exception e) {
+            J1939_84.getLogger().log(Level.SEVERE, "Error loading map from slots", e);
+        }
+        return slots;
+    }
+
+    private final int id;
+
+    private final int length; // bits
+
+    private final String name;
+
+    private final Double offset;
+
+    private final Double scaling;
+
+    private final String type;
+
+    private final String unit;
+
+    private Slot(int id, String name, String type, Double scaling, Double offset, String unit, int length) {
+        this.id = id;
+        this.name = name;
+        this.type = type;
+        this.scaling = scaling;
+        this.offset = offset;
+        this.unit = unit;
+        this.length = length;
+    }
+
+    /**
+     * Returns the Identifier of the SLOT
+     *
+     * @return the id
+     */
+    public int getId() {
+        return id;
+    }
+
+    /**
+     * Returns the length of the data in bits
+     *
+     * @return the length
+     */
+    public int getLength() {
+        return length;
+    }
+
+    /**
+     * Returns the Name of the SLOT
+     *
+     * @return the name
+     */
+    public String getName() {
+        return name;
+    }
+
+    /**
+     * Returns the offset
+     *
+     * @return the offset
+     */
+    public Double getOffset() {
+        return offset;
+    }
+
+    /**
+     * Returns the scaling
+     *
+     * @return the scaling
+     */
+    public Double getScaling() {
+        return scaling;
+    }
+
+    /**
+     * Returns the Type of the SLOT
+     *
+     * @return the type
+     */
+    public String getType() {
+        return type;
+    }
+
+    /**
+     * Returns the Units of the SLOT
+     *
+     * @return the unit
+     */
+    public String getUnit() {
+        return unit;
+    }
+
+    /**
+     * Returns a scaled value. That is result = value * scaling + offset
+     *
+     * @param value the value to scale
+     * @return double
+     */
+    public double scale(double value) {
+        if (getScaling() != null && getOffset() != null) {
```

Comment:
Should probably be
double v = value;
if (getScaling()!=null)v*=getScaling();
if(getOffset()!=null)v+=getOffset();
return v;